integreat_cms/cms/utils/linkcheck_utils/linkcheck_utils.py

```diff
@@ -243,69 +209,43 @@ def replace_links(
-                    )
-                    if should_replace:
-                        fixed_url = url.replace(search, replace)
-                        new_translation.content = rewrite_links(
-                            new_translation.content,
-                            partial(replace_link_helper, url, fixed_url),
-                        )
-                        logger.debug(
-                            "Replacing %r with %r in %r", url, fixed_url, translation
-                        )
-                new_translation.content = fix_content_link_encoding(
-                    new_translation.content
-                )
-                if new_translation.content != translation.content and commit:
-                    save_new_version(translation, new_translation, user)
+        for translation in translations:
+            translation.replace_links(search, replace, links_to_replace, user, commit)
+
     # Wait until all post-save signals have been processed
     logger.debug("Waiting for linkcheck listeners to update link database...")
     time.sleep(0.1)
     tasks_queue.join()
     logger.info("Finished replacing %r with %r in content links", search, replace)
 
 
-def fix_domain_encoding(url: re.Match[str]) -> str:
-    """
-    Fix the encoding of punycode domains
-
-    :param url: The input url match
-    :return: The fixed url
+def find_links_and_translations(
+    search: str, region: Region | None, link_types: list[str] | None
+) -> tuple[list[Link], QuerySet]:
     """
-    parsed_url: ParseResult = urlparse(url.group(1))
-    parsed_url = parsed_url._replace(netloc=unquote(parsed_url.netloc))
-    return parsed_url.geturl()
+    Collects link objects which match the selected link types and contain string given as search in the url,
+    and find translations which contain those links
 
+    :param search: The (partial) URL to search
+    :param region: Optionally limit the replacement to one region (``None`` means a global replacement)
+    :param link_types: Which kind of links should be replaced
 
-def fix_content_link_encoding(content: str) -> str:
+    :return: A tuple of links and translations
     """
-    Fix the encoding of punycode domains in an html content string
+    # This function is used in replace_links, which is used in the management command, where region can be None, too.
+    # However get_region_links currently requires a valid region.
+    # Collect all the link objects in case no region is given.
+    links = get_region_links(region) if region else Link.objects.all()
+    links_to_replace = [
+        link
+        for link in links
+        if search in link.url.url and (not link_types or link.url.type in link_types)
+    ]
+    translations = {link.content_object for link in links_to_replace}
 
-    :param content: The input content
-    :return: The fixed content
-    """
-    return re.sub(r"(?<=[\"'])(https?://.+?)(?=[\"'])", fix_domain_encoding, content)
+    return links_to_replace, translations
```

Comment:
If you agree that it makes sense to change the signature of replace_links, then we should probably also change find_links_and_translations (Also we should change the name, because it will not just return translations in the future). I think it could just return a list of objects, plus their links that should be replaced. In the future we could even try to do this in completely in the database, but that should probably be another pr. So grouping the links by object in python is probably fine for now.

Here is how it could roughly look like:
```python
    links = (get_region_links(region) if region else Link.objects.all()).filter(url__url__contains=search).select_related("url")
    if link_types:
        links_to_replace = [link for link in links if link.url.type in link_types]
    else:
        links_to_replace = links

    content_objects = defaultdict(set)
    for link in links_to_replace:
        content_objects[link.content_object].add(link.url)

    return content_objects
```